uwblueprint/supportive-housing
# Navigation Bar Component

## Notion ticket link

[Frontend Page Skeleton Component](https://github.com/uwblueprint/supportive-housing/issues/41)



## Implementation description
* Created a Navigation Bar component
* Created new pages (that are empty), "Resident Directory" and "Employee Directory"
* Updated Homepage to include Log Records table
* Removed auth requirement and email verification from logout so that users do not need to be verified to log out



## Steps to test
1. Verify the Navigation Bar renders, and is styled correctly according to the Figma
2. Verify all the nav bar buttons work:
    a. Resident Directory takes you to `/resident-directory` endpoint
    b. Employee Directory takes you to `/employee-directory` endpoint
    c. Home takes you to `/` endpoint
    d. Logout signs you out, and takes you the sign out page 
    
Note: In testing, I would often run into I/O errors which would cause the page to not render. If you run into this issue, the (painful) solution is to rebuild the deployment:
```
docker compose down 
docker rm -f $(docker ps -a -q) # deletes all running/stopped containers
docker compose down --volumes # brings down all volumes (incl database)
docker volume rm $(docker volume ls -q) # deletes all docker volumes
docker-compose up --build
docker exec -it SHOW-backend /bin/bash -c "flask db upgrade"
```

## Checklist
